CriteoPublisherSdk/Tests/IntegrationTests/CR_ProfileIdFunctionalTests.m

```diff
@@ -176,24 +177,22 @@ - (void)testStandaloneNative_GivenAnyPreviousIntegration_UseStandaloneProfileId
 
 #pragma mark - InHouse
 
-- (void)prepareInHouseTest:(CRAdUnit *)adUnit {
-  [self prepareCriteoForGettingBidWithAdUnits:@[ adUnit ]];
-  [self enrichAdObject:NSMutableDictionary.new forAdUnit:adUnit];
-
-  [self resetCriteo];
-  [self prepareCriteoForGettingBidWithAdUnits:@[ adUnit ]];
-  [self.criteo.testing_networkCaptor clear];
-}
-
 - (void)testInHouse_GivenAnyPreviousIntegration_UseInHouseProfileId {
   CRBannerAdUnit *adUnit = [CR_TestAdUnits preprodBanner320x50];
+  [self prepareCriteoForGettingBidWithAdUnits:@[ adUnit ]];
+  CRBid *bid = [self getBidWithAdUnit:adUnit];
 
-  [self prepareInHouseTest:adUnit];
+  NSDictionary *request = [self cdbRequest];
+  XCTAssertEqualObjects(request[@"profileId"], @(CR_IntegrationFallback));
 
-  [self.criteo getBidForAdUnit:adUnit];
-  [self waitForIdleState];
+  CRBannerView *bannerView = [[CRBannerView alloc] initWithAdUnit:adUnit criteo:self.criteo];
+  [bannerView loadAdWithBid:bid];  // declares integration
```

Comment:
Could you add test for the other format: interstitial and native ?

Reply:
Done